scripts/data/client.py

```diff
@@ -102,124 +110,132 @@ def process_batch(job):
+            process.terminate()
+            try:
+                process.wait(timeout=2)
+            except subprocess.TimeoutExpired:
+                process.kill()
+            return None, None, -1
+
+    stdout, stderr = process.communicate()
+    return stdout, stderr, process.returncode
+
+
+def process_batch(job):
+    if shutdown_event.is_set():
+        logger.info(f"Shutdown detected, skipping job: {job}")
+        return
+
+    arguments_file = job.batch_file.as_posix().replace(".json", "-arguments.json")
+    try:
+        with open(arguments_file, "w") as af:
+            af.write(str(format_args(job.batch_file, False, False)))
+
+        stdout, stderr, returncode = run_process(arguments_file)
+
+        if stdout is None:  # Process was terminated due to shutdown
+            return
+
+        if (
+            returncode != 0
+            or "FAIL" in stdout
+            or "error" in stdout
+            or "panicked" in stdout
+        ):
+            error = stdout or stderr
+            if returncode == -9:
+                match = re.search(r"gas_spent=(\d+)", stdout)
+                gas_info = (
+                    f", gas spent: {int(match.group(1))}"
+                    if match
+                    else ", no gas info found"
+                )
+                error = f"Return code -9, killed by OOM?{gas_info}"
+                message = error
+            else:
+                error_match = re.search(r"error='([^']*)'", error)
+                message = error_match.group(1) if error_match else ""
+            logger.error(f"{job} error: {message}")
+            logger.debug(f"Full error while processing: {job}:\n{error}")
+        else:
+            match = re.search(r"gas_spent=(\d+)", stdout)
             gas_info = (
-                f", gas spent: {int(match.group(1))}"
-                if match
-                else ", no gas info found"
+                f"gas spent: {int(match.group(1))}" if match else "no gas info found"
             )
-            error = f"Return code -9, killed by OOM?{gas_info}"
-            message = error
-        else:
-            error_match = re.search(r"error='([^']*)'", error)
-            message = error_match.group(1) if error_match else ""
+            logger.info(f"{job} done, {gas_info}")
+            if not match:
+                logger.warning(f"{job}: no gas info found")
 
-        logger.error(f"{job} error: {message}")
-        logger.debug(f"Full error while processing: {job}:\n{error}")
-    else:
-        match = re.search(r"gas_spent=(\d+)", result.stdout)
-        gas_info = f"gas spent: {int(match.group(1))}" if match else "no gas info found"
-        logger.info(f"{job} done, {gas_info}")
-        if not match:
-            logger.warning(f"{job}: not gas info found")
+    except Exception as e:
+        logger.error(f"Error processing batch {job}: {e}")
 
 
-# Producer function: Generates data and adds jobs to the queue
 def job_producer(job_gen):
```

Comment:
Why do you remove comments?

Reply:
An oversight from my end, reverted to the previous form with the comment in the latest fix.